include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -255,18 +254,93 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 			}
 		}
 
+		//string to be replaced with all "#" except those in "#include"
+		static constexpr const char* PREPROC_DIRECTIVE_DISABLER = "_this_is_a_hash_";
+		static constexpr const char* PREPROC_DIRECTIVE_ENABLER = PREPROC_DIRECTIVE_DISABLER;
+
+		// TODO make sure there are no GLSL specifics for these functions in the HLSL compiler
+		// 
+		//all "#", except those in "#include"/"#version"/"#pragma shader_stage(...)", replaced with `PREPROC_DIRECTIVE_DISABLER`
+		static void disableAllDirectivesExceptIncludes(std::string& _code)
```

Comment:
move this and the functions below to cpp

Reply:
they're used in both the glsl and hlsl compiler, that's why i placed it back here

Reply:
there are some glsl specific ones which i'll place in the cpp file for that

Reply:
I meant in `IShaderCompiler.cpp` because why not. I don't want the recompilation to take long by changing a simple function here (since it's in the header)